src/java.desktop/share/classes/javax/swing/border/EtchedBorder.java

```diff
@@ -131,22 +151,59 @@ public EtchedBorder(int etchType, Color highlight, Color shadow)    {
      * @param height the height of the painted border
      */
     public void paintBorder(Component c, Graphics g, int x, int y, int width, int height) {
-        int w = width;
-        int h = height;
+        // We remove any initial transforms to prevent rounding errors
+        // when drawing in non-integer scales
+        AffineTransform at = new AffineTransform();
+        Stroke oldStk = new BasicStroke();
+        int stkWidth = 1;
```

Comment:
Now the usage of these variables is restricted to the case where `resetTransform` is set to `true`. Therefore, they can be left uninitialised here; a value is assigned before it's used in all the code paths as far as I can see.

It's just a small (premature) optimisation, it avoids creating two objects. I don't insist though.

Reply:
I'd be inclined to remove the initialisation.

Reply:
I see, the static analyser is not smart enough to notice that the variables get initialised before they're used.

To resolve the error, initialise `at` and `oldStk` with `null`.

`stkWidth` needs to be initialised by 1 as it's currently done, it's always used.

Reply:
Yes, seems the code logic is too much for the compiler
intuitively I'd have expected that it should be, but you'd need to restructure to fix it ..